### PR TITLE
Documentation

### LINKED ISSUE #88: Readthedocs
Or similar. I think the project is now at a point where we need a dedicated documentation website, where we can also put some cookbooks up (e.g., the example notebooks - compare [scipy cookbooks](https://scipy-cookbook.readthedocs.io/)).

### PR DESCRIPTION
**Pull request by N-Wouda**
Closes #88.

- [ ] Update the notebook links in the top-level README, and point to the docs.
- [x] Add VNS to docs

### CODE REVIEW

**Comment by N-Wouda** [comment]
The documentation's at: https://alns--95.org.readthedocs.build/en/95/api.html

It's not all perfect yet, but it's a good start that we can build on in later PRs.

**Comment by N-Wouda** [comment]
@leonlan docs work now, I think this is ready for a review.

**Comment by N-Wouda** [comment]
TODO: update the links to the example notebooks in the top-level README file.

File: docs/source/alns.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ALNS
+====
+
+The top-level :mod:`alns` module contains two public classes: :class:`~alns.ALNS.ALNS` and :class:`~alns.State.State`.
+The first is used to run the ALNS algorithm.
+The second can be subclassed to define a solution state: all that is needed is to define a `objective()` member function that returns the solution cost.
+
+Finally, the :meth:`~alns.ALNS.ALNS.iterate` method on :class:`~alns.ALNS.ALNS` instances returns a :class:`~alns.Result.Result` object.
+Its properties and methods can be used to access the final solution and runtime statistics.
+
```
Comment (leonlan) [inline docs/source/alns.rst:14]: Shouldn't `alns.Statistics` be included here & in the automodule generation below?

File: docs/source/other single-trajectory heuristics.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+   def greedy_randomised_repair(sol: State, rnd_state) -> State:
+       <do greedy randomised repair around sol>
+       return <improved solution>
+
+
+   alns = ALNS()
+   alns.add_destroy_operator(destroy)
+   alns.add_repair_operator(greedy_randomised_repair)
+
+We again suggest to use :class:`~alns.select.RouletteWheel`, and leave the choice of acceptance and stopping criterion are left to the user.
```
Comment (leonlan) [inline docs/source/other single-trajectory heuristics.rst:81]:
```suggestion
We again suggest to use :class:`~alns.select.RouletteWheel`, and leave the choice of acceptance and stopping criterion to the user.
```

File: docs/source/select.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+   :synopsis: Operator selection schemes
+
+
+Operator selection schemes
+==========================
+
+The :mod:`alns.select` module contains the various operator selection schemes the `alns` package ships with.
+These are used during the ALNS search to select a destroy and repair operator pair in each iteration.
+
+All stopping criteria inherit from :class:`~alns.select.SelectionScheme.SelectionScheme`, which can be subclassed to create your own.
```
Comment (leonlan) [inline docs/source/select.rst:11]:
```suggestion
All operator selection schemes inherit from :class:`~alns.select.SelectionScheme.SelectionScheme`, which can be subclassed to create your own.
```

**Review by leonlan — COMMENTED** [review]
Docs look good!

File: docs/source/accept.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+   :synopsis: Acceptance criteria
+
+
+Acceptance criteria
+===================
+
+The :mod:`alns.accept` module contains the various acceptance criteria the `alns` package ships with.
+These criteria are used by the ALNS algorithm to decide whether to accept or reject a candidate solution.
+
+All acceptance criteria inherit from :class:`~alns.accept.AcceptanceCriterion.AcceptanceCriterion`, which can be subclassed to create your own.
```
[the remark this answers is not in the document]
Comment (N-Wouda) [reply docs/source/accept.rst:11]: Possibly? I'll freely admit my ignorance: I do not really know how Sphinx works yet. This seems to do what I want it to do (provide a link to the class), but there might be a shortcut I haven't yet discovered.

**Comment by N-Wouda** [comment]
@leonlan I think we're mostly there now.

File: docs/source/accept.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (leonlan) [reply docs/source/accept.rst:11]: OK, I don't know how Sphinx works, but maybe we can figure out how to do this later once we know more about it.

**Review by leonlan — APPROVED** [review]

File: docs/source/accept.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (N-Wouda) [reply docs/source/accept.rst:11]: That's what I'm thinking as well. For now this works well enough, and we can always revisit the docs later on!

**Comment by N-Wouda** [comment]
I'll add in VNS and update the readme tomorrow, and then merge this in.

**Comment by N-Wouda** [comment]
Something seems to be wrong with the network/files for Py 3.10. The other CI checks pass, however, so I think this can be merged.

**Comment by N-Wouda** [comment]
Just updated the readme in 60f223bc1dbf276a5e22b829ab0bdb2d504945f8.

**Pull request merged**